apps/examples/security_test/security_test_main.c

```diff
@@ -0,0 +1,1597 @@
+	return 0;
+
+exit:
+	mbedtls_x509_crt_free(&crt);
+	return ret;
+}
+
+unsigned int g_cert_existence[1][2] = { {FACTORYKEY_ARTIK_CERT, 0}
+};
+
+unsigned int g_key_existence[1][2] = { {FACTORYKEY_ARTIK_DEVICE, 0}
+};
+
+void print_existence(void)
+{
+	unsigned char cert[1][20] = { "ARTIK CERT" };
+
+	unsigned char key[1][20] = { "ARTIK DEV" };
+
+	int i;
+
+	for (i = 0; i < 1; i++) {
+		if (g_cert_existence[i][1]) {
+			printf("%s", "[ O ]");
+		} else {
+			printf("%s", "[ X ]");
+		}
+		printf(" %s\n", cert[i]);
+	}
+	for (i = 0; i < 1; i++) {
+		if (g_key_existence[i][1]) {
+			printf("%s", "[ O ]");
+		} else {
+			printf("%s", "[ X ]");
+		}
+		printf(" %s\n", key[i]);
+	}
+}
+
+void init_existence(void)
+{
+	int i;
+
+	for (i = 0; i < 1; i++) {
+		g_cert_existence[i][1] = 0;
+	}
+	for (i = 0; i < 1; i++) {
+		g_key_existence[i][1] = 0;
+	}
+}
+
+void set_existence(unsigned int id)
+{
+	int i;
+
+	for (i = 0; i < 1; i++) {
+		if (g_cert_existence[i][0] == id) {
+			g_cert_existence[i][1] = 1;
+			return;
+		}
+	}
+	for (i = 0; i < 1; i++) {
+		if (g_key_existence[i][0] == id) {
+			g_key_existence[i][1] = 1;
+			return;
+		}
+	}
+}
+
+void TC_see_factory_getcert(int debug_mode)
+{
+	int ret = 0, i = 0;
+	struct fac_info cert[SEE_CERTNUM] = { {FACTORYKEY_ARTIK_CERT, "ARTIK CERT"} };
+
+	unsigned char buf[4096];
+	unsigned int buf_len = 4096;
+
+	while (i < SEE_CERTNUM) {
+		printf("[%d] %s ... ", i, cert[i].name);
+		buf_len = 4096;
+
+		ret = see_get_certificate(buf, &buf_len, cert[i].id, 0);
+		if (ret != 0) {
+			printf("get failed : %d\n", ret);
+			i++;
+			continue;
+		}
+		ret = print_crt(debug_mode, buf, buf_len);
+
+		if (ret) {
+			printf("parse failed : %x\n", -ret);
+			i++;
+			continue;
+		}
+		set_existence(cert[i].id);
+		i++;
+		printf("success\n");
+	}
+}
+
+void TC_see_factory_ecctest(int debug_mode, unsigned int index, unsigned int curve)
+{
+	int i = 0;
+	int ret = 0;
+
+	struct sECC_SIGN ecc_sign;
+	struct sECC_KEY ecc_key;
+
+	printf("--------------- sign/verify start --------------\n");
+
+	unsigned int input[SEE_HASHNUM] = { OID_SHA1_160, OID_SHA2_256,
+					    OID_SHA2_384, OID_SHA2_512
+					  };
+	unsigned int hash_len[SEE_HASHNUM] = { 20, 32, 48, 64 };
+	unsigned char hash[64];
+
+	unsigned char r[68];
+	unsigned char s[68];
+
+	memset(&ecc_sign, 0, sizeof(struct sECC_SIGN));
+
+	ecc_sign.r = r;
+	ecc_sign.s = s;
+
+	while (i < SEE_HASHNUM) {
+		printf("[%d] ARTIK ECC key test (hashlen %d) ...\n", i, hash_len[i]);
+
+		ecc_sign.sign_type = curve | input[i];
+		ret = see_get_ecdsa_signature(&ecc_sign, hash, hash_len[i], index);
+		if (ret != 0) {
+			printf("sign failed : %d\n", ret);
+			break;
+		}
+		printf("sign success\n");
+
+		ret = see_verify_ecdsa_signature(&ecc_sign, hash, hash_len[i], index);
+		if (ret != 0) {
+			printf("verify failed : %d\n", ret);
+			break;
+		}
+		printf("verify success\n");
+
+		i++;
+	}
+
+	while (!ret) {
+		printf("[%d] GET ARTIK PUBKEY ... ", i);
```

Comment:
Is this an ARTIK specific example?

Reply:
Currently yes, but it will be generalized later..